src/App.jsx

```diff
@@ -14,33 +15,78 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
+export const App = () => {
+  const [goodsToSort, setGoodsToSort] = useState(goodsFromServer);
+  const [selectedOption, setSelectedOption] = useState("");
+  const [isReverseActive, setIsReverseActive] = useState(false);
+
+
+  function handleGoodsAlphabetically() {
+      setSelectedOption("Sort alphabetically")
+      if(isReverseActive) {
+        setGoodsToSort([...goodsToSort].sort((a, b) => b.localeCompare(a)));
+      } else{
+        setGoodsToSort([...goodsToSort].sort((a, b) => a.localeCompare(b)));
+      }
+
+  }
+
+  function handleSortGoodsByLength() {
+    setSelectedOption('Sort by length')
+    if(isReverseActive) {
+      setGoodsToSort([...goodsToSort].sort((a, b) => b.length - a.length));
+    } else {
+    setGoodsToSort([...goodsToSort].sort((a, b) =>  a.length - b.length));
+    }
+  }
+
+  function handleReverseGoods() {
+    setGoodsToSort([...goodsToSort].reverse());
+    setIsReverseActive(!isReverseActive);
+  }
+
+  function handleGoodsToOriginal() {
+    setGoodsToSort(goodsFromServer);
+    setSelectedOption("");
+    setIsReverseActive(false);
+  }
+
+  function checkIfArrayIsEqualsToFromServer() {
+    return goodsToSort.every((good, index) => good === goodsFromServer[index]);
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+      <div className="buttons">
```

Comment:
There is a duplicate `<div className="buttons">` tag here, which is unnecessary and could lead to incorrect nesting or styling issues. Consider removing one of these duplicate divs to clean up the JSX structure.